### DISCUSSION TITLE
[Feature Request] date-input accept other formats

### DISCUSSION THREAD

**Discussion by sergiorbj** [heroui-inc/heroui#4743] [Feature requests]
### Is your feature request related to a problem? Please describe.

Currently, the DateInput component only accepts dates in the "mm/dd/yyyy" format. I would like to create a new property that accepts a string that changes this pattern, such as "dd/mm/yyyy".

### Describe the solution you'd like

In DateInput component, put new property to set the date format, like:

`
<DateInput
          format={"dd/mm/yyyy"}
/>
`

### Describe alternatives you've considered

1. Input a property that accepts a string.
2. Input a property that accepts some variants like `"dd/mm/yyyy" | "mm/dd/yyyy"`

**Comment by linear** [comment]
<p><a href="https://linear.app/nextui-inc/issue/ENG-1026/[feature-request]-date-input-accept-other-formats">ENG-1026 [Feature Request] date-input accept other formats</a></p>

**Comment by predragnikolic** [comment]
Would it be able to accept custom format as `dd/mm` or just `yyyy`? for example, this code `<DateInput format={"dd/mm"} />` would allow the user to enter day and month?
